lib/parser/source/buffer.rb

```diff
@@ -178,6 +181,23 @@ def raw_source=(input)
         end
 
         @source = input.gsub("\r\n".freeze, "\n".freeze).freeze
+
+        if defined?(Encoding) &&
+           !@source.ascii_only? &&
+           @source.encoding != Encoding::UTF_32LE
+          begin
+            @slice_source = @source.encode(Encoding::UTF_32LE)
+          rescue EncodingError
```

Comment:
Too defensive. This shouldn't be silently swallowed here, in fact any encoding problem that may arise here should have been caught in `reencode_string`.